File: Codes/Bingran Li/LinkList/206. Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prevNode, currentNode, nextNode;
        prevNode = null;
        currentNode = head;
        while(currentNode!=null){
            nextNode = currentNode.next;
            currentNode.next=prevNode;
            prevNode= currentNode;
            currentNode = nextNode;
        }
        head= prevNode;
        return head;
    }
}

File: Codes/Kan Zheng/C++/linkedlist/206.Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* Node=NULL;
        ListNode* tempNode=NULL;
        ListNode* verseNode=NULL;
        for(Node=head;Node!=NULL;Node=Node->next)
        {
            verseNode = new ListNode(Node->val);
            if(Node!=head)
                verseNode->next = tempNode;
            tempNode = verseNode;
        }
        return verseNode;
    }
};

